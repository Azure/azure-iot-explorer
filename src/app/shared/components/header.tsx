/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { ResourceKeys } from '../../../localization/resourceKeys';
import { SettingsPane } from '../../settings/components/settingsPane';
import '../../css/_header.scss';

export const Header: React.FC = () => {
    const { t } = useTranslation();

    return (
        <header>
            <div className="title">{t(ResourceKeys.header.applicationName)}</div>
            <div className="settings"><SettingsPane/></div>
        </header>
    );
};
