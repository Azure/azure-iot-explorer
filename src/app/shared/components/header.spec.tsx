/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { shallow } from 'enzyme';
import { Header } from './header';

describe('Header', () => {
    it('matches snapshot', () => {

        expect(shallow(<Header/>)).toMatchSnapshot();
    });
});
