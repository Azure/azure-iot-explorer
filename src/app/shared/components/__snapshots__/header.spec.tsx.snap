// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared/components/header matches snapshot when settings is not visible 1`] = `
<header
  className="header"
>
  <div
    className="title"
  >
    header.applicationName
  </div>
  <div>
    <Connect(NotificationList) />
  </div>
  <CustomizedDefaultButton
    ariaLabel="header.settings.launch"
    className=""
    iconProps={
      Object {
        "iconName": "PlugConnected",
      }
    }
    id="settings"
    onClick={[Function]}
    text="header.settings.launch"
  />
</header>
`;

exports[`shared/components/header matches snapshot when settings is visible 1`] = `
<header
  className="header"
>
  <div
    className="title"
  >
    header.applicationName
  </div>
  <div>
    <Connect(NotificationList) />
  </div>
  <CustomizedDefaultButton
    ariaLabel="header.settings.launch"
    className="settings-visible"
    iconProps={
      Object {
        "iconName": "PlugDisconnected",
      }
    }
    id="settings"
    onClick={[Function]}
    text="header.settings.launch"
  />
</header>
`;
